python/tchannel/tornado/inbound_server.py

```diff
@@ -12,77 +9,25 @@
 from tchannel.tornado.connection import TornadoConnection
 
 
-class _ServerRequestAdapter():
-    """Adapts the `TChannelMessageDelegate` interface to the interface expected
-    by our clients.
-    """
-    def __init__(self, server, request_conn, server_conn=None):
-        self.server = server
-        self.connection = request_conn
-        self.request = None
-        if isinstance(server.request_callback,
-                      httputil.HTTPServerConnectionDelegate):
-            self.delegate = server.request_callback.start_request(
-                server_conn, request_conn)
-            self._chunks = None
-        else:
-            self.delegate = None
-            self._chunks = []
-
-    def headers_received(self, start_line, headers):
-        # TODO implement xheaders
-        if self.delegate is None:
-            self.request = httputil.HTTPServerRequest(
-                connection=self.connection, start_line=start_line,
-                headers=headers)
-        else:
-            return self.delegate.headers_received(start_line, headers)
-
-    def data_received(self, chunk):
-        if self.delegate is None:
-            self._chunks.append(chunk)
-        else:
-            return self.delegate.data_received(chunk)
-
-    def finish(self):
-        if self.delegate is None:
-            self.request.body = b''.join(self._chunks)
-            self.request._parse_body()
-            self.server.request_callback(self.request)
-        else:
-            self.delegate.finish()
-        self._cleanup()
-
-    def on_connection_close(self):
-        if self.delegate is None:
-            self._chunks = None
-        else:
-            self.delegate.on_connection_close()
-        self._cleanup()
-
-    def _cleanup(self):
-        # TODO cleanup work
-        pass
-
-
 class InboundServer(tornado.tcpserver.TCPServer):
-    def __init__(self, request_callback=None):
-        self.request_callback = request_callback
+    def __init__(self, req_handler, options=None):
+        assert req_handler is not None
+
+        self.req_handler = req_handler
+        self.options = options
+        self.peers = {}
         super(InboundServer, self).__init__()
 
     def build_stream(self):
```

Comment:
this is a function will be called from tornado iostream. when it accepts a new connection, it will call this method.
It also gives us the chance to save a new tornado connection instance